docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,115 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
+
+**s2n_config_set_psk_selection_callback** sets the **s2n_psk_selection_callback**. If it is not set, the s2n-tls server chooses the first identity in its PSK list that also appears in the client's PSK list.
+
+**s2n_offered_psk_new** creates a new offered PSK object. Pass this object to **s2n_offered_psk_list_next** to retrieve the next PSK from the list.  Use **s2n_offered_psk_list_has_next** prior to this API call to ensure we have not reached the end of the list. **s2n_offered_psk_free** frees the memory associated with the **s2n_offered_psk** object.
+
+**s2n_offered_psk_list_reread** returns the offered PSK list to its original read state. After **s2n_offered_psk_list_reread** is called, the next call to **s2n_offered_psk_list_next** will return the first PSK in the offered PSK list.
+
+**s2n_offered_psk_get_identity** gets the identity and identity length for a given offered PSK object.
+
+**s2n_offered_psk_list_choose_psk** sets the chosen offered PSK to be used for the connection. To disable PSKs for the connection and perform a full handshake instead, set the PSK identity to NULL. 
+
+In the following example, **s2n_psk_selection_callback** chooses the first client offered PSK identity present in an external store.
+
+```c
+int s2n_psk_selection_callback(struct s2n_connection *conn, void *context,
+                               struct s2n_offered_psk_list *psk_list)
+{
+    struct s2n_offered_psk *offered_psk = s2n_offered_psk_new();
+
+    while (s2n_offered_psk_list_has_next(psk_list)) {
+        uint8_t *client_psk_id = NULL;
+        uint16_t client_psk_id_len = 0;
+
+        s2n_offered_psk_list_next(psk_list, offered_psk);
+        s2n_offered_psk_get_identity(offered_psk, &client_psk_id, &client_psk_id_len);
+        struct s2n_psk *psk = user_lookup_identity_db(client_psk_id, client_psk_id_len);
+        
+        if (psk) {
+            s2n_connection_append_psk(conn, psk);
+            s2n_offered_psk_list_choose_psk(psk_list, offered_psk);
+            break;
+        }
+    }
+    s2n_offered_psk_free(&offered_psk);
+    return S2N_SUCCESS;
+}
+```
+
+### Retrieve the Negotiated Pre-Shared Key
+
+The following APIs enable the caller to retrieve the PSK selected by the server for the connection.
+
+```c
+int s2n_connection_get_negotiated_psk_identity_length(struct s2n_connection *conn, uint16_t *identity_length);
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity, uint16_t max_identity_length);
+```
+
+**s2n_connection_get_negotiated_psk_identity** gets the identity of the PSK used to negotiate the connection. **s2n_connection_get_negotiated_psk_identity_length** gets the length of the identity. If the connection performed a full handshake instead of using PSKs **s2n_connection_get_negotiated_psk_identity_length** returns 0 and **s2n_connection_get_negotiated_psk_identity** does nothing.
```

Comment:
```suggestion
**s2n_connection_get_negotiated_psk_identity** gets the identity of the PSK used to negotiate the connection. **s2n_connection_get_negotiated_psk_identity_length** gets the length of the identity. If the connection performed a full handshake instead of using PSKs then **s2n_connection_get_negotiated_psk_identity_length** returns 0 and **s2n_connection_get_negotiated_psk_identity** does nothing.
```